components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -55,9 +57,15 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
   /* ----------------------- WARNING --------------------------------*/
 
   //Now need to read in the file
-  scorpio::register_file(m_datafiles[0],scorpio::Read);
-  m_num_src_levs = scorpio::get_dimlen(m_datafiles[0],"lev");
-  scorpio::eam_pio_closefile(m_datafiles[0]);
+  if (m_src_pres_type == TIME_DEPENDENT_3D_PROFILE) {
+    scorpio::register_file(m_datafiles[0],scorpio::Read);
+    m_num_src_levs = scorpio::get_dimlen(m_datafiles[0],"lev");
+    scorpio::eam_pio_closefile(m_datafiles[0]);
```

Comment:
Actually if it's time dependent, it should never be hard-coded to the first one right?

Reply:
The number of levels should not change with time, only the pressure values. So using the first one should be fine.